projects/exui-common-lib/src/lib/services/feature-toggle/feature-toggle.guard.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, UrlTree } from '@angular/router';
+import { combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { FeatureToggleService } from './feature-toggle.service';
+
+@Injectable()
+export class FeatureToggleGuard implements CanActivate {
+
+    public constructor(
+        private readonly featureToggleService: FeatureToggleService,
+        private readonly router: Router
+    ) {}
+
+    /**
+     * Usage: Add the FeatureToggleGuard as the Guard for a route, and provide the following
+     * in the data array for the route:
+     * - needsFeaturesEnabled: An array of feature keys that need to be enabled for this route
+     * - featureDisabledRedirect: the URL to redirect to when the this route is not accessible due to disabled features
+     * @param route Automatically provided by Angular
+     */
+    public canActivate(route: ActivatedRouteSnapshot): Observable<boolean | UrlTree> {
+        return combineLatest(...(route.data.needsFeaturesEnabled as string[]).map(
```

Comment:
Where is route.data.needsFeaturesEnabled initialised? I couldn't see any code in this repo that does it

Reply:
It is initialized in the route configuration block, where also the URL, component and guard are set for a route. See the comment above the method.